addons/mail/models/mail_thread.py

```diff
@@ -1762,6 +1762,13 @@ def _mail_find_partner_from_emails(self, emails, records=None, force_create=Fals
             followers = records.mapped('message_partner_ids')
         else:
             followers = self.env['res.partner']
+        if records and records._fields.get('company_id'):
```

Comment:
it's a little risky to hardcode things

since this is really generic code that could be used in any custom module, it might be a little risky to just do the code if there is a field named "company_id" but is not a Many2one res.company field (even if it would be very risky in itself to do that).

If it was an issue, I don't see any fix in that besides doing an override only for the model we want. A way to do that would be (if the partner comes from `_mail_search_on_partner`) to override `_mail_search_on_partner` and just order it in the override in the wanted models.

Reply:
[`company_id` is a considered a reserved field name](https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html#odoo.models.Model.company_id), is if someone were to use it like that, they're most likely already getting errors due to all multi-company consistency checks relying on it.